### PR TITLE
feat(hypershift): Priority class dynamic adjustment

### PR DESCRIPTION
**Pull request by a-dsouza**
This is an enhancement document for the priority class issue: https://github.com/openshift/hypershift/issues/1041

supersedes https://github.com/openshift/enhancements/pull/1257

### CODE REVIEW

File: enhancements/hypershift/priority-class/priority-class-dynamic-adjustment.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  - Clients should be able to opt out of employing the features offered through this enhancement.  
+- Define sensible priority class recommendations.
+  - ETCD should always have a higher priority class to account for unwarranted evictions.
+
+### Non-Goals
+- HO will not adjust any pod resource request and limits.
+- Priority class adjustments to a lower priority value will not be supported.
+
+## Proposal
+HyperShift currently creates a static set of default priority classes. We propose to add support for dynamically adjusting priority classes for the underlying control plane workloads. This will help to ensure the HCs with a higher level of usage/investment will be of more importance when scheduling workloads and mitigating preemptions.
```
Comment (jewzaam) [inline enhancements/hypershift/priority-class/priority-class-dynamic-adjustment.md:62]: Does this imply dynamic creation of priority class to accomidate arbitrarily large hosted clusters?

**Review by jewzaam — CHANGES_REQUESTED** [review]

**Comment by dhellmann** [comment]
(automated message) This pull request is closed with lifecycle/rotten. It does not appear to be linked to a valid Jira ticket. Should the PR be reopened, updated, and merged? If not, removing the lifecycle/rotten label will tell this bot to ignore it in the future.

**Comment by csrwng** [comment]
@a-dsouza pls reopen this when we're ready to work on it

**Pull request closed without merging**